thread/test/perf_workpool.cpp

```diff
@@ -41,20 +40,36 @@ void* task_async(void*) {
                                std::memory_order_relaxed);
             sem.signal(1);
         }));
-        photon::thread_yield();
     }
     sem.wait(FLAGS_fires);
-    return nullptr;
 }
 
-void* task_sync(void*) {
-    auto start = std::chrono::steady_clock::now();
-    pool->call([&, start] {
-        auto end = std::chrono::steady_clock::now();
-        sum_time.fetch_add(std::chrono::duration_cast<std::chrono::nanoseconds>(end - start).count(),
-                           std::memory_order_relaxed);
-    });
-    return nullptr;
+void task_sync() {
+    for (uint64_t i = 0; i < FLAGS_fires; ++i) {
+        auto start = std::chrono::steady_clock::now();
+        pool->call<photon::PhotonContext>([&, start] {
+            auto end = std::chrono::steady_clock::now();
+            sum_time.fetch_add(std::chrono::duration_cast<std::chrono::nanoseconds>(end - start).count(),
+                               std::memory_order_relaxed);
+        });
+    }
+}
+
+void task_sync_in_std_context() {
+    for (uint64_t i = 0; i < FLAGS_fires; ++i) {
+        auto start = std::chrono::steady_clock::now();
+        pool->call<photon::StdContext>([&, start] {
+            auto end = std::chrono::steady_clock::now();
+            sum_time.fetch_add(std::chrono::duration_cast<std::chrono::nanoseconds>(end - start).count(),
+                               std::memory_order_relaxed);
+
+        });
+    }
+}
+
+static inline size_t get_qps(std::chrono::time_point<std::chrono::steady_clock> start,
+                             std::chrono::time_point<std::chrono::steady_clock> end) {
+    return FLAGS_fires * 1000 * 1000 / std::chrono::duration_cast<std::chrono::microseconds>(end - start).count();
```

Comment:
QPS seems to be limited by the length of queue. This should be avoided in benchmarks.